Run zulu8 fips CI with BCJSSE instead of SunJSSE

jkakavas opened the pull request:
As we figured out in
https://github.com/elastic/elasticsearch/issues/61316#issuecomment-685482708
Azul brings back a lot of changes from JDK 11 to their Zulu8 build
and this means that we can't run this with SunJSSE in FIPS 140 mode.

This change ensures that we configure Zulu8 JDK JVMs in FIPS 140
mode, using the bouncy castle JSSE FIPS provider, instead of the
SunJSSE one ( as we do for the rest of the java 8 JVMs )

Resolves: #61316

jkakavas committed:
Run zulu8 fips CI with BCJSSE

As we figured out in
https://github.com/elastic/elasticsearch/issues/61316#issuecomment-685482708
Azul brings back a lot of changes from JDK 11 to their Zulu8 build
and this means that we can't run this with SunJSSE in FIPS 140 mode.

This change ensures that we configure Zulu8 JDK JVMs in FIPS 140
mode, using the bouncy castle JSSE FIPS provider, instead of the
SunJSSE one ( as we do for the rest of the java 8 JVMs )

Resolves: #61316

elasticmachine commented:
Pinging @elastic/es-security (:Security/Security)

elasticmachine commented:
Pinging @elastic/es-core-infra (:Core/Infra/Build)

rjernst commented on buildSrc/src/main/java/org/elasticsearch/gradle/info/BuildParams.java:
```diff
@@ -89,6 +90,10 @@ public static JavaVersion getRuntimeJavaVersion() {
         return value(runtimeJavaVersion);
     }
 
+    public static String getJvmDetails() {
```
Can you please open this as a separate change so it goes to master as well? We should probably also name it like "runtime jvm details"?

jkakavas replied:
sure thing, makes sense

jkakavas replied:
https://github.com/elastic/elasticsearch/pull/61901

rjernst commented on gradle/fips.gradle:
```diff
@@ -9,8 +9,9 @@ if (BuildParams.inFipsJvm) {
     boolean java8 = BuildParams.runtimeJavaVersion == JavaVersion.VERSION_1_8
-    File fipsSecurity = new File(fipsResourcesDir, "fips_java${java8 ? '8' : ''}.security")
-    File fipsPolicy = new File(fipsResourcesDir, "fips_java${java8 ? '8' : ''}.policy")
+    boolean zulu8 = java8 && BuildParams.jvmDetails.contains("Zulu")
```
Is the zulu policy the exact same as the java 11, or a mix of our existing java 11 and java 8 policies?

jkakavas replied:
The zulu policy and properties are the same as the java 11 with regards to the actual settings and policy rules , but with the format that is applicable for JDK8. Maybe fips_java8 and fips_java_zulu8 are not applicable names here, I will change them to reflect the crypto provider that is in use in each case so that this is clearer

rjernst reviewed (commented):
Just a few comments

jkakavas committed:
Address feedback

Rename security properties and policy files to reflect the security
provider and the JDK version for which they provide configuration

jkakavas committed:
Merge remote-tracking branch 'origin/7.x' into fix-zulu8-fips

rjernst reviewed (approved):
LGTM

mark-vieira reviewed (approved):
LGTM

jkakavas committed:
Merge remote-tracking branch 'origin/7.x' into fix-zulu8-fips

jkakavas committed:
Merge changes from master

jkakavas commented:
will merge in spite of `elasticsearch-ci/packaging-sample-windows` as it is known to be flaky and my commit makes no changes to packaging
